client/src/Components/AverageRating.jsx

```diff
@@ -1,9 +1,112 @@
 import React from 'react';
+import styled from 'styled-components';
 
-const AverageRating = ({ average, ratings }) => (
-  <div>
-    Hello from AverageRating component!
-  </div>
-);
+/* -------------------------  STYLES ------------------------- */
+
+const DivStarRatingsContainer = styled.div`
+  margin: 5px 0 19px;
+  box-sizing: border-box;
+  inline-size: 150px;
+`;
+
+const DivStarRatingsSubContainer = styled.a`
+  display: flex;
+  align-items: center;
+  text-decoration: none;
+  cursor: pointer;
+`;
+
+const SpanNumericalRating = styled.span`
+  font-size: 2.2rem;
+  font-weight: 600;
+  line-height: 1;
+`;
+
+const SpanStarRatingsWrapper = styled.span`
+  flex-direction: column;
+  margin: 2px 0 0 5px;
+  width: 80px;
+`;
+
+const UlStarRating = styled.ul`
+  margin: 0 0 2px;
+  list-style: none;
+  padding: 0;
+  font-size: 12px;
+  float: left;
+  height: 13.4px;
+`;
+
+const SpanStarText = styled.span`
+  font-weight: 400;
+  line-height: 18.2px;
+  font-size: 14px;
+  line-height: normal;
+`;
+
+const IStar = styled.i`
+  font-size: 15px;
+  font-style: normal;
+  color: ${({ starColor }) => starColor ? "#ffc120" : "#e6e7e8;"};
+  font-family: Material Icons;
+`;
+
+/* -------------------------  RENDER ------------------------- */
+
+const AverageRating = ({ average, ratings }) => {
+
+  // Generate colored stars based on passed in average
+  const stars = [];
+  let iconString = '';
+  let starColor;
+  let remaining = average;
+
+  // For each star, loop through and generate a jsx element with a corresponding star color and shape
+  for (let i = 0; i < 5; i += 1) {
+
+    // Determine star shape (filled or filled half)
+    iconString = (remaining >= 1 || (remaining > 0 && remaining.toFixed(1) % 1 >= 0.8))
+      ? "star"
+      : (remaining > 0 && remaining.toFixed(1) % 1 >= 0.3)
+        ? "star_half"
+        : "star";
+
+    /*
+    material-icons string references for stars:
+      star: filled star
+      star_border: outlined star (not used)
+      star_half: outlined half-filled star
+    */
+
+    // Determine color of the star (grey or gold)
+    starColor = remaining.toFixed(1) >= 0.3 ? true : null;
+
+    // Push the generateted star to the stars array for rendering later
+    stars.push(
+      <IStar className="material-icons" key={i} starColor={starColor}>{iconString}</IStar>
+    );
+
+    // Reduce remaining variable before moving onto next star loop
+    remaining = remaining - 1;
+  }
+
+  return (
+    <DivStarRatingsContainer>
+      <DivStarRatingsSubContainer>
+        <SpanNumericalRating>
+          {average.toFixed(1)}
+        </SpanNumericalRating>
+        <SpanStarRatingsWrapper>
+          <UlStarRating>
+            {stars}
+          </UlStarRating>
+          <SpanStarText>
+            {Math.floor(ratings).toFixed(0)} rating{ratings > 1 ? "s" : null}
```

Comment:
I believe that this processing is because the random number generator is creating very small decimal values at the end of integers?  Maybe consider processing all of your data before it enters the database or outside of the return for better separation of concerns.

Reply:
Yeah, most likely the database will be serving up a rating integer with zero decimal places. When I was sending it dummy data, I gave it a few integers with 1 decimal place, and it looked weird. Realistically _(hopefully)_ it'll never have to deal with this case though.

Perhaps a good candidate for a Jest test! 😉